### DISCUSSION TITLE
Animated graphs in Wave

### DISCUSSION THREAD

**Discussion by n-srinidhi** [h2oai/wave#1967] [Q&A] [answered]
Hi! Is it possible to get interactive graphs i.e. animated graphs in Wave? For example, If I have a pie or a donut chart, On hovering over one category/colour, I want it to pop out. Can this be achieved by Wave?

**Comment by mturoci** [comment]
Hey.

Tooltips and other interactions are available in our `plot_card` and `ui.visualization` widgets. As for stat cards (e.g. [pie](https://wave.h2o.ai/docs/widgets/stat/wide_pie_stat/)), tooltips are not available yet - https://github.com/h2oai/wave/issues/1564.

**Reply by n-srinidhi** [reply]
Hi!I meant something like the following:
If this is my original donut chart,

![image](https://github.com/h2oai/wave/assets/117831190/c62468b3-f847-4521-a19e-0f6cda02374e)

On clicking or hovering over it, I would like it to change to this

![image](https://github.com/h2oai/wave/assets/117831190/a903c1b8-1bb7-4d5b-bc08-9b66fa56fbe0)

**Reply by mturoci** [reply, marked as answer]
Yes, I am talking about the same - https://wave.h2o.ai/blog/release-0.21#let-your-plots-come-to-life

**Reply by n-srinidhi** [reply]
Alright. Thanks a lot!

**Reply by n-srinidhi** [reply]
@mturoci  Hi, I'm trying to register click events for my bar chart created using the `ui.visualization` card, But I am not able to raise the event in the `serve(q)` function, inorder to be able to handle it. How do I go about it?

```
 ui.visualization(name='tr_g',
                         data=data('Symbol Value', 5, rows=data_df, pack=True),
                         plot=ui.plot([
                               ui.mark(coord='theta', type='interval', x='=Symbol', y='=Value', stack='auto',
                               y_min=0, color='$amber')
                           ]),
                          interactions=['click','scale_zoom', 'select_marks'] 
                          ),
```

**Reply by mturoci** [reply]
Maybe [this](https://github.com/h2oai/wave/discussions/2089) could help?
